web/client/components/catalog/editor/MainFormUtils.js

```diff
@@ -13,3 +13,11 @@ export const defaultPlaceholder = (service) => {
     }
     return true;
 };
+
+export const checkIfUrlMatchesProtocol = (url = '') => {
```

Comment:
This check needs to match only for the mapstore instances with https.

The expected behaviours is:
- mapstore with `http:` protocol should allow catalog with `http:` or `https:`
- mapstore with `https:` protocol should allow only catalog with `https:`

I would suggets to use an url parser instead of indexOf, usually we use the 'url' library so it could be implemented like this:

```js
import url from 'url';
// ...

export const checkIfUrlMatchesProtocol = (catalogUrl = '') => {
    const { protocol: mapStoreProtocol } = url.parse(window.location.href);
    const { protocol: catalogProtocol } = url.parse(catalogUrl);
    if (mapStoreProtocol === 'https:') {
        return mapStoreProtocol === catalogProtocol;
    }
    return true;
};
```
Please if you are using this code check if it's working as expected because I did not test it properly (maybe also the name of the function needs to change due to the new behaviour)

Reply:
this Works @allyoucanmap